NullPointerException: Attempt to invoke interface method 'int java.util.List.size()' on a null object reference

Posted by Michael J. Farrand on 4 Jul:

```
USER_COMMENT=null
ANDROID_VERSION=6.0.1
APP_VERSION_NAME=1.15
BRAND=samsung
PHONE_MODEL=SAMSUNG-SM-J120AZ
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.upload.ShareActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'int java.util.List.size()' on a null object reference
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3255)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3351)
at android.app.ActivityThread.access$1100(ActivityThread.java:222)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1796)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:158)
at android.app.ActivityThread.main(ActivityThread.java:7230)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'int java.util.List.size()' on a null object reference
at android.os.Parcel.readStringList(Parcel.java:2109)
at fr.free.nrw.commons.Media.<init>(Media.java:238)
at fr.free.nrw.commons.contributions.Contribution.<init>(Contribution.java:76)
at fr.free.nrw.commons.contributions.Contribution$1.createFromParcel(Contribution.java:25)
at fr.free.nrw.commons.contributions.Contribution$1.createFromParcel(Contribution.java:23)
at android.os.Parcel.readParcelable(Parcel.java:2367)
at android.os.Parcel.readValue(Parcel.java:2264)
at android.os.Parcel.readArrayMapInternal(Parcel.java:2614)
at android.os.BaseBundle.unparcel(BaseBundle.java:221)
at android.os.BaseBundle.getBoolean(BaseBundle.java:659)
at android.app.Activity.restoreHasCurrentPermissionRequest(Activity.java:7126)
at android.app.Activity.performCreate(Activity.java:6876)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1136)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3208)
... 9 more
java.lang.NullPointerException: Attempt to invoke interface method 'int java.util.List.size()' on a null object reference
at android.os.Parcel.readStringList(Parcel.java:2109)
at fr.free.nrw.commons.Media.<init>(Media.java:238)
at fr.free.nrw.commons.contributions.Contribution.<init>(Contribution.java:76)
at fr.free.nrw.commons.contributions.Contribution$1.createFromParcel(Contribution.java:25)
at fr.free.nrw.commons.contributions.Contribution$1.createFromParcel(Contribution.java:23)
at android.os.Parcel.readParcelable(Parcel.java:2367)
at android.os.Parcel.readValue(Parcel.java:2264)
at android.os.Parcel.readArrayMapInternal(Parcel.java:2614)
at android.os.BaseBundle.unparcel(BaseBundle.java:221)
at android.os.BaseBundle.getBoolean(BaseBundle.java:659)
at android.app.Activity.restoreHasCurrentPermissionRequest(Activity.java:7126)
at android.app.Activity.performCreate(Activity.java:6876)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1136)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3208)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3351)
at android.app.ActivityThread.access$1100(ActivityThread.java:222)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1796)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:158)
at android.app.ActivityThread.main(ActivityThread.java:7230)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

--- a/app/src/main/java/fr/free/nrw/commons/Media.java
+++ b/app/src/main/java/fr/free/nrw/commons/Media.java
@@ -235,8 +235,9 @@ public class Media implements Parcelable {
         width = in.readInt();
         height = in.readInt();
         license = in.readString();
-        //FIXME: Null check
-        in.readStringList(categories);
+        if (categories != null) {
+            in.readStringList(categories);
+        }
         descriptions = in.readHashMap(ClassLoader.getSystemClassLoader());
     }
 

--- a/app/src/main/java/fr/free/nrw/commons/Media.java
+++ b/app/src/main/java/fr/free/nrw/commons/Media.java
@@ -235,6 +235,7 @@ public class Media implements Parcelable {
         width = in.readInt();
         height = in.readInt();
         license = in.readString();
+        //FIXME: Null check
         in.readStringList(categories);
         descriptions = in.readHashMap(ClassLoader.getSystemClassLoader());
     }

